packages/next-server/lib/router/router.js

```diff
@@ -38,6 +38,7 @@ export default class Router {
     this.subscriptions = new Set()
     this.componentLoadCancel = null
     this._beforePopState = () => true
+    this._beforeRouteChangeStart = () => true
```

Comment:
I'm not sure about this implementation, why can't we just use the eventemitter?

Reply:
We need to prevent the the routing from continuing in the `async change` method.  This is analogous to what `beforePopState` is doing but for general route changes.